controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -35,49 +47,69 @@ import (
 // nolint
 type StorageClassClaimReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	Scheme            *runtime.Scheme
+	storageClassClaim *ocsv1alpha1.StorageClassClaim
+	log               logr.Logger
+	storageCluster    *v1.StorageCluster
+	ctx               context.Context
+	namespace         string
 }
 
+const (
+	StorageClassClaimFinalizer                      = "storageclassclaim.ocs.openshift.io"
+	StorageClassFinalizer                           = "storageclass.ocs.openshift.io"
+	FulfillStorageClassClaimRequestBlockpool        = 0
+	FulfillStorageClassClaimRequestSharedfilesystem = 1
+)
+
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclusters,verbs=get;watch;list
+// +kubebuilder:rbac:groups=storage.k8s.io,resources=storageclasses,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=persistentvolumes,verbs=get;list;watch
 
 func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx, "StorageClassClaim", request)
-	ctx = ctrllog.IntoContext(ctx, log)
-	log.Info("Reconciling StorageClassClaim.")
+	r.ctx = ctx
+	r.log = ctrllog.Log.WithValues("storageclassclaim", request.NamespacedName)
+	r.log.Info("Reconciling StorageClassClaim.")
+	r.namespace = request.Namespace
 
 	// Fetch the StorageClassClaim instance
-	claim := &ocsv1alpha1.StorageClassClaim{}
-	if err := r.Client.Get(ctx, request.NamespacedName, claim); err != nil {
+	r.storageClassClaim = &ocsv1alpha1.StorageClassClaim{}
+	if err := r.get(r.storageClassClaim); err != nil {
```

Comment:
Don't we need to provide name and namespace of storageclassclaim?

Reply:
Fixed.